integration-test/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/KubernetesSuite.scala

```diff
@@ -66,20 +66,61 @@ private[spark] class KubernetesSuite extends FunSuite with BeforeAndAfterAll wit
     runSparkPiAndVerifyCompletion()
   }
 
+  test("Run SparkPi with a master URL without a scheme.") {
+    val url = kubernetesTestComponents.kubernetesClient.getMasterUrl
+    sparkAppConf.set("spark.master", s"k8s://${url.getHost}:${url.getPort}")
+    runSparkPiAndVerifyCompletion()
+  }
+
+  test("Run SparkPi with custom driver pod name.") {
```

Comment:
Likewise for below? Reason to separate this test from below?

Reply:
To be honest, I'm not a fan of having tests that cover multiple aspects related or not related just for reducing the number of test cases.

Reply:
That makes sense. There will just be ALOT of test cases already, and these are merely checking the metaData so they seem somewhat coupled... no?

Reply:
Sure. Done.